pkg/driver/node.go

```diff
@@ -0,0 +1,258 @@
+package driver
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+)
+
+var (
+	nodeCaps       = []csi.NodeServiceCapability_RPC_Type{}
+	fsVolumeType   = "filesystem"
+	volVolumeType  = "volume"
+	rootVolumePath = "fsx"
+)
+
+func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "")
+}
+
+func (d *Driver) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "")
+}
+
+// NodePublishVolume Mounts the PV at the target path.
+func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	klog.V(4).Infof("NodePublishVolume: Called with args %+v", req)
+
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID missing in NodePublishVolumeRequest")
+	}
+
+	volCap := req.GetVolumeCapability()
+	if volCap == nil {
+		return nil, status.Error(codes.InvalidArgument, "Volume capability missing in NodePublishVolumeRequest")
+	}
+	if !d.isValidVolumeCapabilities([]*csi.VolumeCapability{volCap}) {
+		return nil, status.Error(codes.InvalidArgument, "NodePublishVolume: Volume capability not supported")
+	}
+
+	context := req.GetVolumeContext()
+	dnsname := context["dnsname"]
+	mountname := context["mountname"]
+	volumeType := strings.ToLower(context["volumeType"])
```

Comment:
if its a map lookup you might want to check if the value exists, even if there is some sort of static analysis where we would force this its good to validate server side

Reply:
Go defaults non-existent key retrievals to the zero value of the type associated with the map (in this case an empty string). I have validation below these value retrievals to check for this, so I don't think I'd need to do another check here. I also have unit tests ensuring that the behavior is as expected when a user doesn't provide these values.